org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/codelens/LSPCodeMining.java

```diff
@@ -52,14 +53,11 @@ public LSPCodeMining(CodeLens codeLens, IDocument document, LanguageServer langu
 
 	@Override
 	protected CompletableFuture<Void> doResolve(ITextViewer viewer, IProgressMonitor monitor) {
-		if (!LanguageServiceAccessor.checkCapability(languageServer,
-				capabilities -> {
-					Boolean resolveProvider = capabilities.getCodeLensProvider().getResolveProvider();
-					return resolveProvider != null && resolveProvider;
-				})) {
+		if (Boolean.TRUE != this.languageServerWrapper.getServerCapabilities().getCodeLensProvider().getResolveProvider()) {
```

Comment:
I think 
```
		Boolean resolveProvider = capabilities.getCodeLensProvider().getResolveProvider();
					
		if (resolveProvider != null && resolveProvider) {
			return CompletableFuture.completedFuture(null);
		}					
```
would read better.

Reply:
That inverts the logic! I have rewritten it along the lines you suggest, though ;-)